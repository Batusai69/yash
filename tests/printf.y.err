printf: `--no-such-option' is not a valid option
Usage:  printf format [value...]
Usage:  printf format [value...]
