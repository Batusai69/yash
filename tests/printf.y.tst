# printf.y.tst: yash-specific test of the echo and printf builtins
# vim: set ft=sh ts=8 sts=4 sw=4 noet:

echo ===== echo =====

unset ECHO_STYLE

echo 123 456 789
echo 1 22 '3  3' "4
4" 5\
5

testecho() {
    echo ===== ${ECHO_STYLE-unset}
    echo -n new
    echo line
    echo -n new line
    echo '1\a2\b3\c4' 5
    echo '6\f7\n8\r9\t0\v!'
    echo '\0123\012\01x' '\123\12\1x' '\00411'
    echo -e '1\c2'
    echo -e -E '1\c2'
    echo -eE '1\c2'
    echo -ne 123 '-\c-' 456
    echo
}

testecho
ECHO_STYLE=SYSV testecho
ECHO_STYLE=XSI  testecho
ECHO_STYLE=BSD  testecho
ECHO_STYLE=GNU  testecho
ECHO_STYLE=ZSH  testecho
ECHO_STYLE=DASH testecho
ECHO_STYLE=RAW  testecho


echo ===== printf =====

printf ''
printf '1\n'
printf -- 'hello\n' ignored arguments
printf 'multiple\nlines '
printf 'continued\n'

echo =====

printf 'a\ab\bf\fr\rt\tv\v-\n'
printf 'backslash \\\n'
printf 'double-quote \"\n'
printf "single-quote \\'\n"
printf 'octal \11 \1000\n'

echo ===== %%

printf '%%\n'
printf '%%\n' ignored arguments

echo ===== %d

printf '%d\n'
printf '%d\n' 0
printf '%d ' 0 10 200 +345 -1278; echo 1
printf '%+d ' 1 20 +345 -1278; echo 2
printf '%+ d ' 1 20 +345 -1278; echo 3
printf '% d ' 1 20 +345 -1278; echo 4
printf '%+8d ' 1 20 +345 -1278; echo 5
printf '%+-8d ' 1 20 +345 -1278; echo 6
printf '%+08d ' 1 20 +345 -1278; echo 7
printf '%+-08d ' 1 20 +345 -1278; echo 8
printf '%+8.3d ' 1 20 +345 -1278; echo 9
printf '%d ' 00 011 0x1fE 0X1Fe \'a \"@; echo 10

echo ===== %i

printf '%i\n'
printf '%i\n' 0
printf '%i ' 0 10 200 +345 -1278; echo 1
printf '%+i ' 1 20 +345 -1278; echo 2
printf '%+ i ' 1 20 +345 -1278; echo 3
printf '% i ' 1 20 +345 -1278; echo 4
printf '%+8i ' 1 20 +345 -1278; echo 5
printf '%+-8i ' 1 20 +345 -1278; echo 6
printf '%+08i ' 1 20 +345 -1278; echo 7
printf '%+-08i ' 1 20 +345 -1278; echo 8
printf '%+8.3i ' 1 20 +345 -1278; echo 9
printf '%i ' 00 011 0x1fE 0X1Fe \'a \"@; echo 10

echo ===== %u

printf '%u\n'
printf '%u\n' 0
printf '%u ' 0 10 200 +345; echo 1
printf '%+u ' 1 20 +345; echo 2
printf '%+ u ' 1 20 +345; echo 3
printf '% u ' 1 20 +345; echo 4
printf '%8u ' 1 20 +345; echo 5
printf '%-8u ' 1 20 +345; echo 6
printf '%08u ' 1 20 +345; echo 7
printf '%-08u ' 1 20 +345; echo 8
printf '%8.3u ' 1 20 +345; echo 9
printf '%u ' 00 011 0x1fE 0X1Fe \'a \"@; echo 10

echo ===== %o

printf '%o\n'
printf '%o\n' 0
printf '%o ' 0 10 200 +345; echo 1
printf '%+o ' 1 20 +345; echo 2
printf '%+ o ' 1 20 +345; echo 3
printf '% o ' 1 20 +345; echo 4
printf '%8o ' 1 20 +345; echo 5
printf '%-8o ' 1 20 +345; echo 6
printf '%08o ' 1 20 +345; echo 7
printf '%-08o ' 1 20 +345; echo 8
printf '%8.3o ' 1 20 +345; echo 9
printf '%o ' 00 011 0x1fE 0X1Fe \'a \"@; echo 10
printf '%#.3o ' 0 3 010 0123 012345; echo 11

echo ===== %x

printf '%x\n'
printf '%x\n' 0
printf '%x ' 0 10 200 +345; echo 1
printf '%+x ' 1 20 +345; echo 2
printf '%+ x ' 1 20 +345; echo 3
printf '% x ' 1 20 +345; echo 4
printf '%8x ' 1 20 +345; echo 5
printf '%-8x ' 1 20 +345; echo 6
printf '%08x ' 1 20 +345; echo 7
printf '%-08x ' 1 20 +345; echo 8
printf '%8.3x ' 1 20 +345; echo 9
printf '%x ' 00 011 0x1fE 0X1Fe \'a \"@; echo 10
printf '%#x ' 00 011 0x1fE 0X1Fe \'a \"@; echo 11

echo ===== %X

printf '%X\n'
printf '%X\n' 0
printf '%X ' 0 10 200 +345; echo 1
printf '%+X ' 1 20 +345; echo 2
printf '%+ X ' 1 20 +345; echo 3
printf '% X ' 1 20 +345; echo 4
printf '%8X ' 1 20 +345; echo 5
printf '%-8X ' 1 20 +345; echo 6
printf '%08X ' 1 20 +345; echo 7
printf '%-08X ' 1 20 +345; echo 8
printf '%8.3X ' 1 20 +345; echo 9
printf '%X ' 00 011 0x1fE 0X1Fe \'a \"@; echo 10
printf '%#X ' 00 011 0x1fE 0X1Fe \'a \"@; echo 11

echo ===== %f

printf '%f\n'
printf '%f\n' 0
printf '%f ' 1 1.25 -1000.0; echo 1
printf '%+f ' 1 1.25 -1000.0; echo 2
printf '%+ f ' 1 1.25 -1000.0; echo 3
printf '% f ' 1 1.25 -1000.0; echo 4
printf '%+15f ' 1 1.25 -1000.0; echo 5
printf '%+-15f ' 1 1.25 -1000.0; echo 6
printf '%+015f ' 1 1.25 -1000.0; echo 7
printf '%+-015f ' 1 1.25 -1000.0; echo 8
printf '%+15.3f ' 1 1.25 -1000.0; echo 9
printf '%.0f ' 1 1.25 -1000.0; echo 10
printf '%#.0f ' 1 1.25 -1000.0; echo 11

echo ===== %F

printf '%F\n'
printf '%F\n' 0
printf '%F ' 1 1.25 -1000.0; echo 1
printf '%+F ' 1 1.25 -1000.0; echo 2
printf '%+ F ' 1 1.25 -1000.0; echo 3
printf '% F ' 1 1.25 -1000.0; echo 4
printf '%+15F ' 1 1.25 -1000.0; echo 5
printf '%+-15F ' 1 1.25 -1000.0; echo 6
printf '%+015F ' 1 1.25 -1000.0; echo 7
printf '%+-015F ' 1 1.25 -1000.0; echo 8
printf '%+15.3F ' 1 1.25 -1000.0; echo 9
printf '%.0F ' 1 1.25 -1000.0; echo 10
printf '%#.0F ' 1 1.25 -1000.0; echo 11

echo ===== %e

printf '%e\n'
printf '%e\n' 0
printf '%e ' 1 1.25 -1000.0; echo 1
printf '%+e ' 1 1.25 -1000.0; echo 2
printf '%+ e ' 1 1.25 -1000.0; echo 3
printf '% e ' 1 1.25 -1000.0; echo 4
printf '%+15e ' 1 1.25 -1000.0; echo 5
printf '%+-15e ' 1 1.25 -1000.0; echo 6
printf '%+015e ' 1 1.25 -1000.0; echo 7
printf '%+-015e ' 1 1.25 -1000.0; echo 8
printf '%+15.3e ' 1 1.25 -1000.0; echo 9
printf '%.0e ' 1 1.25 -1000.0; echo 10
printf '%#.0e ' 1 1.25 -1000.0; echo 11

echo ===== %E

printf '%E\n'
printf '%E\n' 0
printf '%E ' 1 1.25 -1000.0; echo 1
printf '%+E ' 1 1.25 -1000.0; echo 2
printf '%+ E ' 1 1.25 -1000.0; echo 3
printf '% E ' 1 1.25 -1000.0; echo 4
printf '%+15E ' 1 1.25 -1000.0; echo 5
printf '%+-15E ' 1 1.25 -1000.0; echo 6
printf '%+015E ' 1 1.25 -1000.0; echo 7
printf '%+-015E ' 1 1.25 -1000.0; echo 8
printf '%+15.3E ' 1 1.25 -1000.0; echo 9
printf '%.0E ' 1 1.25 -1000.0; echo 10
printf '%#.0E ' 1 1.25 -1000.0; echo 11

echo ===== %g

printf '%g\n'
printf '%g\n' 0
printf '%g ' 1 1.25 -1000.0 0.000025; echo 1
printf '%+g ' 1 1.25 -1000.0 0.000025; echo 2
printf '%+ g ' 1 1.25 -1000.0 0.000025; echo 3
printf '% g ' 1 1.25 -1000.0 0.000025; echo 4
printf '%+15g ' 1 1.25 -1000.0 0.000025; echo 5
printf '%+-15g ' 1 1.25 -1000.0 0.000025; echo 6
printf '%+015g ' 1 1.25 -1000.0 0.000025; echo 7
printf '%+-015g ' 1 1.25 -1000.0 0.000025; echo 8
printf '%+15.3g ' 1 1.25 -1000.0 0.000025; echo 9
printf '%+#15.3g ' 1 1.25 -1000.0 0.000025; echo 10

echo ===== %G

printf '%G\n'
printf '%G\n' 0
printf '%G ' 1 1.25 -1000.0 0.000025; echo 1
printf '%+G ' 1 1.25 -1000.0 0.000025; echo 2
printf '%+ G ' 1 1.25 -1000.0 0.000025; echo 3
printf '% G ' 1 1.25 -1000.0 0.000025; echo 4
printf '%+15G ' 1 1.25 -1000.0 0.000025; echo 5
printf '%+-15G ' 1 1.25 -1000.0 0.000025; echo 6
printf '%+015G ' 1 1.25 -1000.0 0.000025; echo 7
printf '%+-015G ' 1 1.25 -1000.0 0.000025; echo 8
printf '%+15.3G ' 1 1.25 -1000.0 0.000025; echo 9
printf '%+#15.3G ' 1 1.25 -1000.0 0.000025; echo 10

echo ===== %c

printf '%c\n'
printf '%c\n' ''
printf '%c\n' a
printf '%c\n' long arguments
printf '%3c\n' b
printf '%-3c\n' c

echo ===== %s

printf '%s\n'
printf '%s\n' 'argument  with  space and
newline'
printf '%s\n' 'argument with backslash \\ and percent %%'
printf '%5s ' 123 a long_argument; echo 1
printf '%5.2s ' 123 a long_argument; echo 2
printf '%-5s ' 123 a long_argument; echo 3

echo ===== %b

printf '%b\n'
printf '%b\n' 'argument  with  space and
newline'
printf '%b\n' 'argument with backslash \\ and percent %%'
printf '%5b ' 123 a long_argument; echo 1
printf '%5.2b ' 123 a long_argument; echo 2
printf '%-5b ' 123 a long_argument; echo 3
printf '%-5.2b ' 123 a long_argument; echo 4
printf '%b\n' '1\a2\b3\c4' 5
printf '%b\n' '6\f7\n8\r9\t0\v!'
printf '%b\n' '\0123\012\01x' '\123\12\1x' '\00411'

echo =====

printf '%d\n' not_a_integer >/dev/null 2>&1
echo $?

echo $(printf '%d\n' not_a_integer 2>/dev/null | wc -l)

echo ===== error =====

printf --no-such-option
echo printf no-such-option $?
(printf foo >&- 2>/dev/null)
echo printf output error $?
printf
echo printf operand missing $?
printf '%d\n' foo 2>/dev/null
echo printf invalid operand $?
(echo foo >&- 2>/dev/null)
echo echo output error $?
