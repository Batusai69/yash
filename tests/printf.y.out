===== echo =====
123 456 789
1 22 3  3 4
4 55
===== unset
-n new
line
-n new line
12367
89	0!
S
x \123\12\1x !1
-e 1-e -E 1-eE 1-ne 123 -
===== SYSV
-n new
line
-n new line
12367
89	0!
S
x \123\12\1x !1
-e 1-e -E 1-eE 1-ne 123 -
===== XSI
-n new
line
-n new line
12367
89	0!
S
x \123\12\1x !1
-e 1-e -E 1-eE 1-ne 123 -
===== BSD
newline
new line1\a2\b3\c4 5
6\f7\n8\r9\t0\v!
\0123\012\01x \123\12\1x \00411
-e 1\c2
-e -E 1\c2
-eE 1\c2
-ne 123 -\c- 456

===== GNU
newline
new line1\a2\b3\c4 5
6\f7\n8\r9\t0\v!
\0123\012\01x \123\12\1x \00411
11\c2
1\c2
123 -
===== ZSH
newline
new line12367
89	0!
S
x \123\12\1x !1
11\c2
1\c2
123 -
===== DASH
newline
new line12367
89	0!
S
x \123\12\1x !1
-e 1-e -E 1-eE 1-ne 123 -
===== RAW
-n new
line
-n new line
1\a2\b3\c4 5
6\f7\n8\r9\t0\v!
\0123\012\01x \123\12\1x \00411
-e 1\c2
-e -E 1\c2
-eE 1\c2
-ne 123 -\c- 456

===== printf =====
1
hello
multiple
lines continued
=====
abfrt	v-
backslash \
double-quote "
single-quote '
octal 	 @0
===== %%
%
%
===== %d
0
0
0 10 200 345 -1278 1
+1 +20 +345 -1278 2
+1 +20 +345 -1278 3
 1  20  345 -1278 4
      +1      +20     +345    -1278 5
+1       +20      +345     -1278    6
+0000001 +0000020 +0000345 -0001278 7
+1       +20      +345     -1278    8
    +001     +020     +345    -1278 9
0 9 510 510 97 64 10
===== %i
0
0
0 10 200 345 -1278 1
+1 +20 +345 -1278 2
+1 +20 +345 -1278 3
 1  20  345 -1278 4
      +1      +20     +345    -1278 5
+1       +20      +345     -1278    6
+0000001 +0000020 +0000345 -0001278 7
+1       +20      +345     -1278    8
    +001     +020     +345    -1278 9
0 9 510 510 97 64 10
===== %u
0
0
0 10 200 345 1
1 20 345 2
1 20 345 3
1 20 345 4
       1       20      345 5
1        20       345      6
00000001 00000020 00000345 7
1        20       345      8
     001      020      345 9
0 9 510 510 97 64 10
===== %o
0
0
0 12 310 531 1
1 24 531 2
1 24 531 3
1 24 531 4
       1       24      531 5
1        24       531      6
00000001 00000024 00000531 7
1        24       531      8
     001      024      531 9
0 11 776 776 141 100 10
000 003 010 0123 012345 11
===== %x
0
0
0 a c8 159 1
1 14 159 2
1 14 159 3
1 14 159 4
       1       14      159 5
1        14       159      6
00000001 00000014 00000159 7
1        14       159      8
     001      014      159 9
0 9 1fe 1fe 61 40 10
0 0x9 0x1fe 0x1fe 0x61 0x40 11
===== %X
0
0
0 A C8 159 1
1 14 159 2
1 14 159 3
1 14 159 4
       1       14      159 5
1        14       159      6
00000001 00000014 00000159 7
1        14       159      8
     001      014      159 9
0 9 1FE 1FE 61 40 10
0 0X9 0X1FE 0X1FE 0X61 0X40 11
===== %f
0.000000
0.000000
1.000000 1.250000 -1000.000000 1
+1.000000 +1.250000 -1000.000000 2
+1.000000 +1.250000 -1000.000000 3
 1.000000  1.250000 -1000.000000 4
      +1.000000       +1.250000    -1000.000000 5
+1.000000       +1.250000       -1000.000000    6
+0000001.000000 +0000001.250000 -0001000.000000 7
+1.000000       +1.250000       -1000.000000    8
         +1.000          +1.250       -1000.000 9
1 1 -1000 10
1. 1. -1000. 11
===== %F
0.000000
0.000000
1.000000 1.250000 -1000.000000 1
+1.000000 +1.250000 -1000.000000 2
+1.000000 +1.250000 -1000.000000 3
 1.000000  1.250000 -1000.000000 4
      +1.000000       +1.250000    -1000.000000 5
+1.000000       +1.250000       -1000.000000    6
+0000001.000000 +0000001.250000 -0001000.000000 7
+1.000000       +1.250000       -1000.000000    8
         +1.000          +1.250       -1000.000 9
1 1 -1000 10
1. 1. -1000. 11
===== %e
0.000000e+00
0.000000e+00
1.000000e+00 1.250000e+00 -1.000000e+03 1
+1.000000e+00 +1.250000e+00 -1.000000e+03 2
+1.000000e+00 +1.250000e+00 -1.000000e+03 3
 1.000000e+00  1.250000e+00 -1.000000e+03 4
  +1.000000e+00   +1.250000e+00   -1.000000e+03 5
+1.000000e+00   +1.250000e+00   -1.000000e+03   6
+001.000000e+00 +001.250000e+00 -001.000000e+03 7
+1.000000e+00   +1.250000e+00   -1.000000e+03   8
     +1.000e+00      +1.250e+00      -1.000e+03 9
1e+00 1e+00 -1e+03 10
1.e+00 1.e+00 -1.e+03 11
===== %E
0.000000E+00
0.000000E+00
1.000000E+00 1.250000E+00 -1.000000E+03 1
+1.000000E+00 +1.250000E+00 -1.000000E+03 2
+1.000000E+00 +1.250000E+00 -1.000000E+03 3
 1.000000E+00  1.250000E+00 -1.000000E+03 4
  +1.000000E+00   +1.250000E+00   -1.000000E+03 5
+1.000000E+00   +1.250000E+00   -1.000000E+03   6
+001.000000E+00 +001.250000E+00 -001.000000E+03 7
+1.000000E+00   +1.250000E+00   -1.000000E+03   8
     +1.000E+00      +1.250E+00      -1.000E+03 9
1E+00 1E+00 -1E+03 10
1.E+00 1.E+00 -1.E+03 11
===== %g
0
0
1 1.25 -1000 2.5e-05 1
+1 +1.25 -1000 +2.5e-05 2
+1 +1.25 -1000 +2.5e-05 3
 1  1.25 -1000  2.5e-05 4
             +1           +1.25           -1000        +2.5e-05 5
+1              +1.25           -1000           +2.5e-05        6
+00000000000001 +00000000001.25 -00000000001000 +00000002.5e-05 7
+1              +1.25           -1000           +2.5e-05        8
             +1           +1.25          -1e+03        +2.5e-05 9
          +1.00           +1.25       -1.00e+03       +2.50e-05 10
===== %G
0
0
1 1.25 -1000 2.5E-05 1
+1 +1.25 -1000 +2.5E-05 2
+1 +1.25 -1000 +2.5E-05 3
 1  1.25 -1000  2.5E-05 4
             +1           +1.25           -1000        +2.5E-05 5
+1              +1.25           -1000           +2.5E-05        6
+00000000000001 +00000000001.25 -00000000001000 +00000002.5E-05 7
+1              +1.25           -1000           +2.5E-05        8
             +1           +1.25          -1E+03        +2.5E-05 9
          +1.00           +1.25       -1.00E+03       +2.50E-05 10
===== %c


a
l
a
  b
c  
===== %s

argument  with  space and
newline
argument with backslash \\ and percent %%
  123     a long_argument 1
   12     a    lo 2
123   a     long_argument 3
===== %b

argument  with  space and
newline
argument with backslash \ and percent %%
  123     a long_argument 1
   12     a    lo 2
123   a     long_argument 3
12    a     lo    4
12367
89	0!
S
x
\123\12\1x
!1
=====
1
1
===== error =====
printf no-such-option 2
printf output error 1
printf operand missing 2
0
printf invalid operand 1
echo output error 1
